src/Microsoft.IIS.Administration.Files/Copy/MoveOperation.cs

```diff
@@ -40,26 +39,7 @@ public MoveOperation(IFileInfo source, IFileInfo destination, string tempPath, I
         public IFileInfo Destination { get; private set; }
         public IFileInfo Source { get; private set; }
         public long TotalSize { get; private set; }
-
-        public long CurrentSize {
-            get {
-                if (_currentSize != -1) {
-                    return _currentSize;
-                }
-
-                if (Source.Type == FileType.File) {
-                    // refresh
-                    var dest = string.IsNullOrEmpty(TempPath) ? _fileProvider.GetFile(Destination.Path) : _fileProvider.GetFile(TempPath);
```

Comment:
I read somewhere if you use `FileInfo fi = new FileInfo(path)` it might work, mind trying this?

Reply:
Might work for what? This issue is kind of deep seeded. Whenever we acquire IFileInfo within our service we have to use the IFileProvider for security reasons. The IFileProvider follows symlinks when it acquires a file info. This requires getting a handle to the file. Having this logic therefore causes the file to be in use. Which is improper for move scenarios. Also, this code was kind of useless since it was only being used for moves of a single file.